app/addons/auth/actions.js

```diff
@@ -28,88 +28,133 @@ var errorHandler = function (xhr, type, msg) {
 };
 
 
-export default {
+function login (username, password, urlBack) {
+  var promise = FauxtonAPI.session.login(username, password);
+
+  promise.then(function () {
+    FauxtonAPI.addNotification({ msg: FauxtonAPI.session.messages.loggedIn });
+    if (urlBack) {
+      return FauxtonAPI.navigate(urlBack);
+    }
+    FauxtonAPI.navigate('/');
+  });
+  promise.fail(errorHandler);
+}
 
-  login: function (username, password, urlBack) {
-    var promise = FauxtonAPI.session.login(username, password);
+function changePassword (password, passwordConfirm) {
+  var nodes = nodesStore.getNodes();
+  var promise = FauxtonAPI.session.changePassword(password, passwordConfirm, nodes[0].node);
 
-    promise.then(function () {
-      FauxtonAPI.addNotification({ msg: FauxtonAPI.session.messages.loggedIn });
-      if (urlBack) {
-        return FauxtonAPI.navigate(urlBack);
-      }
-      FauxtonAPI.navigate('/');
-    });
-    promise.fail(errorHandler);
-  },
+  promise.then(() => {
+    FauxtonAPI.addNotification({ msg: FauxtonAPI.session.messages.changePassword });
+    FauxtonAPI.dispatch({ type: ActionTypes.AUTH_CLEAR_CHANGE_PWD_FIELDS });
+  });
 
-  changePassword: function (password, passwordConfirm) {
-    var nodes = nodesStore.getNodes();
-    var promise = FauxtonAPI.session.changePassword(password, passwordConfirm, nodes[0].node);
+  promise.fail(errorHandler);
```

Comment:
We mustn't use `promise.fail` we cant to eventually move away from jquery and use something else.